src/quantum/perfherder.jsx

```diff
@@ -76,49 +107,62 @@ export default class PerfherderWidget extends React.Component {
         .y0(d => yScale(d.q1))
         .y1(d => yScale(d.q3))
         .curve(curveLinear);
-
       const long = evolutions.length > 3;
       const $evolutions = evolutions.map((evolution, idx) => {
         if (!evolution) {
           return null;
         }
-        const ref = evolution.find(d => moment(d.time * 1000).format('YYYY-MM-DD') === referenceTime);
+
+        const ref = evolution.find(
+          d => moment(d.time * 1000).format('YYYY-MM-DD') === referenceTime
+        );
         let $reference = null;
+
         if (ref) {
           const refY = yScale(ref.avg);
+
           referenceYs.push(refY);
           $reference = busy ? null : (
             <line
               key={`reference-${idx}`}
-              className='reference reference-x'
+              className="reference reference-x"
```

Comment:
Same as earlier comment - see if you can find an attribute of the evolutions data that can be used as a key instead of the index.

Reply:
There was nothing unique in the `evolution` which could be used except the date. So i did it.